Source/CompilationDVDView.cpp

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2010-2017, BurnItNow Team. All rights reserved.
+ * Distributed under the terms of the MIT License.
+ */
+#include "CompilationDVDView.h"
+#include "CommandThread.h"
+
+#include <Alert.h>
+#include <Catalog.h>
+#include <ControlLook.h>
+#include <FindDirectory.h>
+#include <LayoutBuilder.h>
+#include <Path.h>
+#include <ScrollView.h>
+#include <String.h>
+#include <StringView.h>
+
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "DVD view"
+
+#include <compat/sys/stat.h>
+
+static const float kControlPadding = be_control_look->DefaultItemSpacing();
+
+// Message constants
+const int32 kChooseDirectoryMessage = 'Cusd';
+const int32 kBurnerMessage = 'Brnr';
+const int32 kBuildImageMessage = 'IMAG';
+const int32 kBurnDiscMessage = 'BURN';
+const int32 kNoPathMessage = 'Noph'; // defined in PathView
+
+
+CompilationDVDView::CompilationDVDView(BurnWindow& parent)
+	:
+	BView(B_TRANSLATE("DVD-Video"), B_WILL_DRAW,
+		new BGroupLayout(B_VERTICAL, kControlPadding)),
+	fOpenPanel(NULL),
+	fBurnerThread(NULL),
+	fDirPath(new BPath()),
+	fImagePath(new BPath())
+{
+	windowParent = &parent;
+	step = 0;
+
+	SetViewColor(ui_color(B_PANEL_BACKGROUND_COLOR));
+
+	fBurnerInfoBox = new BSeparatorView(B_HORIZONTAL, B_FANCY_BORDER);
+	fBurnerInfoBox->SetFont(be_bold_font);
+	fBurnerInfoBox->SetLabel(B_TRANSLATE_COMMENT(
+		"Choose DVD Video folder to burn", "Status notification"));
+
+	fPathView = new PathView("FolderStringView",
+		B_TRANSLATE("Folder: <none>"));
+
+	fBurnerInfoTextView = new BTextView("DVDInfoTextView");
+	fBurnerInfoTextView->SetWordWrap(false);
+	fBurnerInfoTextView->MakeEditable(false);
+	BScrollView* infoScrollView = new BScrollView("DVDInfoScrollView",
+		fBurnerInfoTextView, B_WILL_DRAW, true, true);
+	infoScrollView->SetExplicitMinSize(BSize(B_SIZE_UNSET, 64));
+
+	fDVDButton = new BButton("ChooseDVDButton",
+		B_TRANSLATE("Choose DVD folder"),
+		new BMessage(kChooseDirectoryMessage));
+	fDVDButton->SetTarget(this);
+	fDVDButton->SetExplicitMaxSize(BSize(B_SIZE_UNLIMITED,
+		B_SIZE_UNSET));
+		
+	fImageButton = new BButton("BuildImageButton", B_TRANSLATE("Build image"),
+	    new BMessage(kBuildImageMessage));
+	fImageButton->SetTarget(this);
+	fImageButton->SetExplicitMaxSize(BSize(B_SIZE_UNLIMITED, B_SIZE_UNSET));
+
+	fBurnButton = new BButton("BurnImageButton", B_TRANSLATE("Burn disc"),
+		new BMessage(kBurnDiscMessage));
+	fBurnButton->SetTarget(this);
+	fBurnButton->SetExplicitMaxSize(BSize(B_SIZE_UNLIMITED, B_SIZE_UNSET));
+
+	BLayoutBuilder::Group<>(dynamic_cast<BGroupLayout*>(GetLayout()))
+		.SetInsets(kControlPadding)
+		.AddGroup(B_HORIZONTAL)
+			.Add(fPathView)
+			.AddGlue()
+			.AddGroup(B_HORIZONTAL)
+				.Add(fDVDButton)
+				.Add(fImageButton)
+				.Add(fBurnButton)
+				.End()
+			.End()
+		.AddGroup(B_VERTICAL)
+			.Add(fBurnerInfoBox)
+			.Add(infoScrollView)
+			.End();
+}
+
+
+CompilationDVDView::~CompilationDVDView()
+{
+	delete fBurnerThread;
```

Comment:
fOpenPanel should be deleted in the destructor.

Reply:
I think you're right (well, knowing you, I'm sure you're right...).
Should be done for the other "tab" source files as well.